internal/servers/controller/handlers/host_catalogs/host_catalog_service.go

```diff
@@ -38,6 +65,33 @@ type Service struct {
 
 var _ pbs.HostCatalogServiceServer = Service{}
 
+func populateCollectionAuthorizedActions(ctx context.Context,
+	authResults auth.VerifyResults,
+	item *pb.HostCatalog) error {
+	resource := &perms.Resource{
+		ScopeId: authResults.Scope.Id,
+		Pin:     item.Id,
+	}
+	// Range over the defined collections and check permissions against those
+	// collections. We use the ID of this scope being returned, not its parent,
+	// hence passing in a resource here.
+	for k, v := range collectionTypeMap {
+		resource.Type = k
+		acts := authResults.FetchActionsForType(ctx, k, v, auth.WithResource(resource)).Strings()
+		if len(acts) > 0 {
+			if item.AuthorizedCollectionActions == nil {
+				item.AuthorizedCollectionActions = make(map[string]*structpb.ListValue)
+			}
+			lv, err := structpb.NewList(strutil.StringListToInterfaceList(acts))
+			if err != nil {
+				return err
+			}
+			item.AuthorizedCollectionActions[k.String()+"s"] = lv
```

Comment:
I'm ok with this merging as is, but just wanted to call out this feels a bit hacky - might be nice to have a function to converts this to a print-friendly version.  This could also lead to some confusion if someone tries to consume this map at some point in the future

Reply:
That's fair. I could make the method return a list of pre-pluralized collections but I wasn't sure that was always what we might want from it. I agree it's hacky but it felt like returning anything other than the actual resource string might assume too much.

Reply:
Going to leave as-is for the moment.